src/private_cuda.h

```diff
@@ -128,4 +140,14 @@ struct _gpukernel {
 #endif
 };
 
+struct _gpucomm {
+  cuda_context* ctx;  // Start after the context
+// #ifdef WITH_CUDA_NCCL
+  ncclComm_t c;
+// #endif  // WITH_CUDA_NCCL
```

Comment:
You could move it to the gpucomm struct and make gpucomm_error take a gpucomm instead.  But that would break the central thing with gpucontext_error so I don't know.

I would like to avoid forcing users to create giant switches based on the error code to get an accurate error description in any case.  Maybe having a place to store a string in the gpucontext struct that would be the error message for the last error?